### PR TITLE
Add substrate to configuration crate

### PR DESCRIPTION
**Pull request by luketchang**
## Motivation

Agents substrate integration requires changes to `configuration` crate (since currently centered around eth contract model).

## Solution
Add knowledge of substrate to `configuration` crate. Leave rest of configuration steps (in `nomad-base`) unimplemented until #232 
- Add additional config called `configs/testMultiVm.json` which has outbound connections to other chains but no inbound connections (no substrate replica support)
- Change contract info (previously CoreContracts and BridgeContracts) to be CoreDeploymentInfo and BridgeDeploymentInfo, with support for substrate variant
- Update TS definitions for wasm_bindgen

## PR Checklist

- [x] Added Tests
- [x] Updated Documentation
- [x] Updated CHANGELOG.md for the appropriate package
- [ ] Ran PR in local/dev/staging

### CODE REVIEW

File: configuration/configs/test.json [hunk 1]
```diff
@@ -273,7 +273,7 @@
   "agent": {
     "ethereum": {
       "rpcStyle": "ethereum",
-      "db": "/usr/share/nomad",
+      "db": "db",
```
Comment (lattejed) [inline configuration/configs/test.json:276]: Need more context on this

**Review by lattejed — APPROVED** [review]
Should we be updating watcher CHANGELOG as well?

File: configuration/configs/test.json (hunk 1 above)
Comment (luketchang) [reply configuration/configs/test.json:276]: Yeah just wanted to change this to `db` so we can run the script locally (as I am doing for local testing). `/usr/share` is a protected folder you can't write to without mounting or smth non-local-friendly

**Comment by luketchang** [comment]
> Should we be updating watcher CHANGELOG as well?
No given change is just adding an `unimplemented!()` stub probably not important

**Comment by lattejed** [comment]
> > Should we be updating watcher CHANGELOG as well?
> 
> No given change is just adding an `unimplemented!()` stub probably not important

Sounds good

**Pull request merged**